osu.Game.Resources/Icons/Win32Icon.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Reflection;
+
+namespace osu.Game.Resources.Icons
+{
+    public class Win32Icon
+    {
+        public readonly Assembly Assembly;
+        public readonly int Id;
+
+        /// <summary>
+        /// String that can be used with the <c>DefaultIcon</c> key in the registry to point to this icon.
+        /// </summary>
+        // The icon id needs to be negative, see https://devblogs.microsoft.com/oldnewthing/20100505-00/?p=14153
+        public string RegistryString => $"{Assembly.Location},{-Id}";
```

Comment:
So my primary question here would be.......... can we skip all of this legacy nonsense and just use plain ICO files?

I'd prefer that we didn't have to use some win32 resource incantations and some executables that are only available on windows and also only if you have visual studio enabled.

I would much rather put the `.ico` assets directly in `osu.Game.Desktop` and point the registry entries at a plain ICO file rather than deal with whatever this thing is. The fact that this is having to cite a Raymond Chen blogpost is a _red flag_ in my eyes in that this is underdocumented legacy cruft, because if _that_ is the best available citation then _oh boy_.

Reply:
In fact as I can tell even microsoft doesn't wanna deal with this stuff anymore:

![image](https://github.com/ppy/osu-resources/assets/20418176/efc7eae5-7077-4052-b354-e96855dd3f3d)

The above is a VS Code `.md` file association. It points at a plain `.ico`.

Reply:
> I would much rather put the .ico assets directly in osu.Game.Desktop and point the registry entries at a plain ICO file 

I agree, would make things simpler. 

I tried including the icons as `Content` and it seems to work well. I just tested this with [velopack](https://github.com/velopack/velopack?tab=readme-ov-file#getting-started) and a sample app, and the icons are available in the install directory.

```xml
<ItemGroup>
    <Content Include="Icons\*.ico"
             CopyToPublishDirectory="PreserveNewest"
             CopyToOutputDirectory="PreserveNewest" />
</ItemGroup>
```

```
C:\Users\...\AppData\Local\VPTest> tree /F
│   Update.exe
│
├───current
│   │   Microsoft.Extensions.DependencyInjection.Abstractions.dll
│   │   Microsoft.Extensions.Logging.Abstractions.dll
│   │   NuGet.Versioning.dll
│   │   sq.version
│   │   Velopack.dll
│   │   VelopackTest.deps.json
│   │   VelopackTest.dll
│   │   VelopackTest.exe
│   │   VelopackTest.runtimeconfig.json
│   │
│   └───Icons
│           lazer.ico
│
└───packages
        VPTest-1.0.0-full.nupkg
```